values.go

```diff
@@ -2102,6 +2102,11 @@ func (v Function) Type() Type {
 	return v.FunctionType
 }
 
+func (v Function) WithType(functionType *FunctionType) Function {
+	v.FunctionType = functionType
+	return v
+}
+
```

Comment:
@turbolent  Thanks for adding new tests and refactoring existing tests!  Great work!  :+1: 

Only item I spotted is that I think adding `Function.WithType()` isn't necessary because `Function` is created with `FunctionType`.

```
func NewFunction(functionType *FunctionType) Function {
	return Function{
		FunctionType: functionType,
	}
}
```

Also, test data for `Function` may need to be adjusted for the same reason.

Reply:
Good point, reverted adding it in f45e8d26d073e07b3238c961493126b518e1cd9a